metrics/metrics.go

```diff
@@ -127,537 +127,603 @@ const (
 
 	TiKVGrpcConnectionState = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "grpc_connection_state",
-			Help:      "State of gRPC connection",
+			Namespace:   namespace,
+			Subsystem:   subsystem,
+			Name:        "grpc_connection_state",
+			Help:        "State of gRPC connection",
+			ConstLabels: constLabels,
 		}, []string{"connection_id", "store_ip", "grpc_state"})
 
 	TiKVAggressiveLockedKeysCounter = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "aggressive_locking_count",
-			Help:      "Counter of keys locked in aggressive locking mode",
+			Namespace:   namespace,
+			Subsystem:   subsystem,
+			Name:        "aggressive_locking_count",
+			Help:        "Counter of keys locked in aggressive locking mode",
+			ConstLabels: constLabels,
 		}, []string{LblType})
 
 	TiKVStoreSlowScoreGauge = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "store_slow_score",
-			Help:      "Slow scores of each tikv node based on RPC timecosts",
+			Namespace:   namespace,
+			Subsystem:   subsystem,
+			Name:        "store_slow_score",
+			Help:        "Slow scores of each tikv node based on RPC timecosts",
+			ConstLabels: constLabels,
 		}, []string{LblStore})
 
 	TiKVPreferLeaderFlowsGauge = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "prefer_leader_flows_gauge",
-			Help:      "Counter of flows under PreferLeader mode.",
+			Namespace:   namespace,
+			Subsystem:   subsystem,
+			Name:        "prefer_leader_flows_gauge",
+			Help:        "Counter of flows under PreferLeader mode.",
+			ConstLabels: constLabels,
 		}, []string{LblType, LblStore})
 
 	TiKVSyncRegionDuration = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "sync_region_duration",
-			Help:      "Bucketed histogram of the duration of sync region.",
-			Buckets:   prometheus.ExponentialBuckets(0.0005, 2, 28),
+			Namespace:   namespace,
+			Subsystem:   subsystem,
+			Name:        "sync_region_duration",
+			Help:        "Bucketed histogram of the duration of sync region.",
+			ConstLabels: constLabels,
+			Buckets:     prometheus.ExponentialBuckets(0.0005, 2, 28),
 		}, []string{LblType})
 
 	initShortcuts()
 }
 
 func init() {
-	initMetrics("tikv", "client_go")
+	initMetrics("tikv", "client_go", nil)
 }
 
 // InitMetrics initializes metrics variables with given namespace and subsystem name.
 func InitMetrics(namespace, subsystem string) {
-	initMetrics(namespace, subsystem)
+	initMetrics(namespace, subsystem, nil)
+}
+
+// InitMetricsWithConstLabels initializes metrics variables with given namespace, subsystem name and const labels.
+func InitMetricsWithConstLabels(namespace, subsystem string, constLabels prometheus.Labels) {
```

Comment:
When will we call this function to init metrics?

Reply:
TiDB will call this function with const labels. If TiDB configures a keyspace when it starts, the client-go's metrics will have keyspace label.